### PR TITLE
Upgrade `eslint-config-colony` to v8

### PR DESCRIPTION
**Pull request by chmanie**
## Description

This has to wait for TS to land. Also this will yield hundreds of errors, that's why I'm waiting for it to be transferred, then fix the errors.

**Changes** 🏗

* A bunch of `eslint`-related packages upgraded

## TODO

- [x] Wait for #1725 to be done
- [x] https://www.npmjs.com/package/@typescript-eslint/eslint-plugin
- [x] https://www.npmjs.com/package/@typescript-eslint/parser
- [x] Fix all the errors

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,6 +57,8 @@
         "@babel/preset-typescript": "^7.3.3",
         "@babel/register": "^7.5.5",
         "@colony/colony-js-adapter": "^1.14.0-beta.2",
+        "@colony/eslint-config-colony": "^8.0.0",
+        "@kironeducation/flow-junit-transformer": "^0.3.0",
```
Comment (JamesLefrere) [inline package.json]: How is this package used?

File: webpack.config.js [hunk 2]
```diff
@@ -168,6 +168,9 @@ const config = {
     new webpack.ProvidePlugin({
       regeneratorRuntime: '@babel/runtime/regenerator',
     }),
+    new webpack.WatchIgnorePlugin([
```
Comment (JamesLefrere) [inline webpack.config.js]: Clever!

File: .eslintrc.js [hunk 3]
```diff
@@ -62,39 +60,10 @@ module.exports = {
...
-      {
-        js: 'off',
-        tsx: 'off',
-        json: 'always',
-      },
-    ],
+    'eslint-comments/disable-enable-pair': ['error', { allowWholeFile: true }],
 
     // @typescript-eslint
+    '@typescript-eslint/ban-ts-ignore': 'off',
```
Comment (JamesLefrere) [inline .eslintrc.js]: Maybe one day :earnest-thinking:

File: src/styles/shared/layout.css [hunk 4]
```diff
@@ -33,57 +33,6 @@
   align-items: center;
 }
 
-.flexContent {
```
Comment (JamesLefrere) [inline src/styles/shared/layout.css]: I presume this was unused - good job 👍

File: package.json (hunk 1 above)
Comment (chmanie) [reply package.json]: I have no idea why this is in there. I’ll fix

Comment (chmanie) [reply package.json]: Done

**Comment by chmanie** [comment]
I downgraded typescript to 3.5.3 until eslint supports the new version (messed with my IDE)

**Pull request merged**